core/server/middleware/middleware.js

```diff
@@ -111,7 +111,7 @@ middleware = {
         });
 
         if (subPath.indexOf('/ghost/api/') === 0
-            && path.indexOf('/ghost/api/v0.1/authentication/') !== 0) {
+            && (path.indexOf('/ghost/api/v0.1/authentication/') !== 0 || req.method === 'PUT')) {
```

Comment:
The way to test it would be making a request PUT to the `authentication/setup` endpoint via something like POSTMAN or curl without a bearer token - it should not allow you to make the request.

As for code testing inside the app, I don't think there is any testing of the authentication middleware at present, this should be done as part of #5286.